backend.pm

```diff
@@ -179,6 +179,24 @@ sub run_cmd
 	wantarray? ($ret, @log) : $ret;
 }
 
+sub run_cmds
+{
+	my ($self, $cmds,%args) = @_;
+	my @log;
+
+	push(@log, {cmd=>'', ret=>0, log=>''}) unless (@{$cmds});
+
+	for my $cmd (@{$cmds}) {
+		my @ret = run_cmd($self, $cmd, $args{timeout});
+		my @output = @ret[1 .. $#ret];
+		push(@log, {cmd=>$cmd, ret=>$ret[0], log => \@output});
+
+		return wantarray? @log : $ret[0]
+		unless (defined($ret[0]) && $ret[0] == 0);
```

Comment:
Maybe it's just me being perl beginner, but I do find this construct a bit confusing, you can easily miss that there is no semicolon at the end of the return line and wonder what this is supposed to do.